openstack/sharedfilesystems/v2/shares/requests.go

```diff
@@ -85,3 +83,41 @@ func GetExportLocations(client *gophercloud.ServiceClient, id string) (r GetExpo
 	_, r.Err = client.Get(getExportLocationsURL(client, id), &r.Body, nil)
 	return
 }
+
+// GrantAccessOptsBuilder allows extensions to add additional parameters to the
+// GrantAccess request.
+type GrantAccessOptsBuilder interface {
+	ToGrantAccessMap() (map[string]interface{}, error)
+}
+
+// GrantAccessOpts contains the options for creation of an GrantAccess request.
+// For more information about these parameters, please, refer to the shared file systems API v2,
+// Share Actions, Grant Access documentation
+type GrantAccessOpts struct {
+	// The access rule type that can be "ip", "cert" or "user".
+	AccessType string `json:"access_type"`
+	// The value that defines the access that can be a valid format of IP, cert or user.
+	AccessTo string `json:"access_to"`
+	// The access level to the share is either "rw" or "ro".
+	AccessLevel string `json:"access_level"`
+}
+
+// ToGrantAccessMap assembles a request body based on the contents of a
+// GrantAccessOpts.
+func (opts GrantAccessOpts) ToGrantAccessMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "allow_access")
+}
+
+// GrantAccess will grant access to a Share based on the values in GrantAccessOpts. To extract
+// the GrantAccess object from the response, call the Extract method on the GrantAccessResult.
+func GrantAccess(client *gophercloud.ServiceClient, id string, opts GrantAccessOptsBuilder) (r GrantAccessResult) {
+	b, err := opts.ToGrantAccessMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	_, r.Err = client.Post(grantAccessURL(client, id), b, &r.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{200, 201},
```

Comment:
Do you know when a 201 would be returned? Have you seen a 201 in any `manila --debug` output? I have only been able to find a default 200 response [here](https://github.com/openstack/manila/blob/stable/newton/manila/api/openstack/wsgi.py#L369).

These return codes are usually the hardest pieces to find. I'm OK leaving both 200 and 201 here unless you're confident it's safe to drop 201.